app/src/main/java/com/kickstarter/viewmodels/projectpage/AddOnsViewModel.kt

```diff
@@ -229,10 +228,14 @@ class AddOnsViewModel(val environment: Environment, bundle: Bundle? = null) : Vi
             holder[it.id()] = it
         }
 
-        // Take the backed AddOns, update with the backed AddOn information which will contain the backed quantity
-        backedAddOns.map {
-            holder[it.id()] = it
-            currentSelection[it.id()] = it.quantity() ?: 0
+        // Take the backed AddOns, update with matching addOn ID with the quantity information
+        backedAddOns.map { backedAddOn ->
+            val aux = holder[backedAddOn.id()]
+            if (aux != null) {
+                val updated = aux.toBuilder().quantity(backedAddOn.quantity()).build()
+                holder[backedAddOn.id()] = updated
```

Comment:
before we were using the information from the baked object, now we update the "quanity" alone to not loose the "shipping rules" field